presto-orc/src/main/java/com/facebook/presto/orc/writer/SliceDictionaryColumnWriter.java

```diff
@@ -178,19 +178,17 @@ public OptionalInt tryConvertToDirect(int maxDirectBytes)
         for (DictionaryRowGroup rowGroup : rowGroups) {
             directColumnWriter.beginRowGroup();
             // todo we should be able to pass the stats down to avoid recalculating min and max
-            writeDictionaryRowGroup(dictionaryValues, rowGroup.getValueCount(), rowGroup.getDictionaryIndexes());
-            directColumnWriter.finishRowGroup();
-
-            if (directColumnWriter.getBufferedBytes() > maxDirectBytes) {
-                directColumnWriter.close();
-                directColumnWriter.reset();
+            if (!writeDictionaryRowGroup(dictionaryValues, rowGroup.getValueCount(), rowGroup.getDictionaryIndexes(), maxDirectBytes)) {
```

Comment:
Do we have an existing test that covers this?

Reply:
@dain : The case of direct conversion exceeds stripe size limit is tested in `testNotDirectConversionOnDictionaryFull`. 

However, when a single row group exceeds 2G after direct conversion, it will fail inside `SliceDictionaryColumnWriter::tryConvertToDirect`. And this is fixed in this commit. 

I will see if we can add a unit test for `SliceDictionaryColumnWriter::tryConvertToDirect`